src/plugins/path/path.c

```diff
@@ -65,6 +90,136 @@ static int validateKey (Key * key, Key * parentKey)
 	return 1;
 }
 
+// I assume the path exists and only validate permission
+static int validatePermission (Key * key, Key * parentKey)
+{
+	uid_t currentUID = geteuid ();
+	gid_t currentGID = getegid ();
+
+	const Key * userMeta = keyGetMeta (key, "check/permission/user");
+	const Key * userTypes = keyGetMeta (key, "check/permission/types");
+
+	// ***** central variables *******
+	const char * validPath = keyString (key);
+	const char * name = keyString (userMeta);
+	const char * modes = keyString (userTypes);
+	// ****************************
+
+	struct passwd * p;
+
+	// Changing to specified user. Can only be done when executing user is root user
+	if (userMeta)
+	{
+		p = getpwnam (name);
+		// Check if user exists
+		if (p == NULL)
+		{
+			ELEKTRA_SET_ERRORF (201, parentKey,
+					    "Could not find user \"%s\" for key \"%s\". "
+					    "Does the user exist?\"",
+					    name, keyName (key));
+			return -1;
+		}
+		name = p->pw_name;
+
+		// Check if I can change the UID as root
+		int err = seteuid ((int) p->pw_uid);
+		if (err < 0)
+		{
+			ELEKTRA_SET_ERRORF (202, parentKey,
+					    "Could not set euid of user \"%s\" for key \"%s\"."
+					    " Are you running kdb as root?\"",
+					    name, keyName (key));
+			return -1;
+		}
+	}
+	else
+	{
+		uid_t uid = geteuid ();
+		p = getpwuid (uid);
+		name = p->pw_name;
+		if (uid != 0)
+		{
+			ELEKTRA_SET_ERRORF (202, parentKey,
+					    "To check permissions for %s I need to be the root user."
+					    " Are you running kdb as root?\"",
+					    keyName (key));
+			return -1;
+		}
+	}
+
+	// The following code changes the egid if a group from a user matches the filegroup
+	// TODO: Whats a good default value for ngroups
+	// If the relevant group is ngroups+1 then it wont get recognized
+	int ngroups = 30;
```

Comment:
Avoid this magic constant.

Reply:
I tried to find another way around but the I did not find a way around the "magic constant".
[getgrouplist](http://man7.org/linux/man-pages/man3/getgrouplist.3.html) requires a number in order to know how many groups to fetch.

Does anybody know a better way?